crates/blockchain/payload.rs

```diff
@@ -444,41 +510,148 @@ fn apply_blob_transaction(
+            context.evm_state,
+            spec_id(
+                &context.chain_config().map_err(ChainError::from)?,
+                context.payload.header.timestamp,
+            ),
+        )?;
+        context.remaining_gas = context.remaining_gas.saturating_sub(report.gas_used());
+        context.block_value += U256::from(report.gas_used()) * head.tip;
+        let receipt = Receipt::new(
+            head.tx.tx_type(),
+            report.is_success(),
+            context.payload.header.gas_limit - context.remaining_gas,
+            report.logs(),
+        );
+        Ok(receipt)
+    }
 }
 
-fn finalize_payload(context: &mut PayloadBuildContext) -> Result<(), StoreError> {
-    let account_updates = get_state_transitions(context.evm_state);
-    // Note: This is commented because it is still being used in development.
-    // dbg!(&account_updates);
-    context.payload.header.state_root = context
-        .store()
-        .ok_or(StoreError::MissingStore)?
-        .apply_account_updates(context.parent_hash(), &account_updates)?
-        .unwrap_or_default();
-    context.payload.header.transactions_root =
-        compute_transactions_root(&context.payload.body.transactions);
-    context.payload.header.receipts_root = compute_receipts_root(&context.receipts);
-    context.payload.header.gas_used = context.payload.header.gas_limit - context.remaining_gas;
-    Ok(())
+fn finalize_payload(
+    context: &mut PayloadBuildContext,
+    block_cache: &mut HashMap<Address, Account>,
+) -> Result<(), StoreError> {
+    #[cfg(feature = "levm")]
+    {
+        if let Some(withdrawals) = &context.payload.body.withdrawals {
+            // For every withdrawal we increment the target account's balance
+            for (address, increment) in withdrawals
+                .iter()
+                .filter(|withdrawal| withdrawal.amount > 0)
+                .map(|w| (w.address, u128::from(w.amount) * u128::from(GWEI_TO_WEI)))
+            {
+                // We check if it was in block_cache, if not, we get it from DB.
+                let mut account = block_cache.get(&address).cloned().unwrap_or({
+                    let acc_info = context
+                        .store()
+                        .unwrap()
+                        .get_account_info_by_hash(context.parent_hash(), address)?
+                        .unwrap_or_default();
+                    let acc_code = context
+                        .store()
+                        .unwrap()
+                        .get_account_code(acc_info.code_hash)?
+                        .unwrap_or_default();
+
+                    Account {
+                        info: AccountInfo {
+                            balance: acc_info.balance,
+                            bytecode: acc_code,
+                            nonce: acc_info.nonce,
+                        },
+                        storage: HashMap::new(), // this must be changed, get the storage from the DB
```

Comment:
What do you think of this:
```suggestion
                        // TODO: this must be changed, get the storage from the DB.
                        // See: https://github.com/lambdaclass/ethrex/issues/1853
                        storage: HashMap::new(),
```

So others seeing the code can have a quick access to the issue?